pulsar-client-go/pulsar/schema.go

```diff
@@ -0,0 +1,298 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+package pulsar
+
+import (
+	"bytes"
+	"encoding/json"
+	"reflect"
+	"unsafe"
+
+	"github.com/gogo/protobuf/proto"
+)
+
+type SchemaType int
+
+const (
+	NONE         SchemaType = iota //No schema defined
+	STRING                         //Simple String encoding with UTF-8
+	INT8                           //A 8-byte integer.
+	INT16                          //A 16-byte integer.
+	INT32                          //A 32-byte integer.
+	INT64                          //A 64-byte integer.
+	FLOAT32                        //A float number.
```

Comment:
it would be good to align what we used in java - FLOAT for FLOAT32 and DOUBLE for FLOAT64